sci-biology/fastqc/fastqc-0.11.3.ebuild

```diff
@@ -1,39 +1,45 @@
-# Copyright 1999-2015 Gentoo Foundation
+# Copyright 1999-2020 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=5
+EAPI=7
 
 inherit java-pkg-2 eutils java-ant-2
 
 DESCRIPTION="Quality control FASTA/FASTQ sequence files"
-HOMEPAGE="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/"
-SRC_URI="http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v"${PV}"_source.zip"
+HOMEPAGE="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/"
+SRC_URI="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v"${PV}"_source.zip"
 
 LICENSE="GPL-3+"
 SLOT="0"
-KEYWORDS=""
-IUSE=""
+KEYWORDS="~amd64"
 
-DEPEND="sci-biology/picard
-	sci-libs/jhdf5
-	>=virtual/jre-1.5:*"
-RDEPEND="${DEPEND}
+RDEPEND="
 	dev-lang/perl
+	>=virtual/jre-1.5:*
+	"
+DEPEND="$RDEPEND
 	>=virtual/jdk-1.5:*
-	dev-java/ant-core"
+	"
 
 S="${WORKDIR}"/FastQC
 
 src_prepare(){
 	cp "${FILESDIR}"/build.xml . || die
+	eapply_user
 }
 
 src_compile(){
 	ant || die
 }
 
 src_install(){
-	dobin fastqc run_fastqc.bat
+	insinto "opt/${PN}"
+	doins -r bin
+	chmod a+x "${ED}/opt/${PN}/bin/fastqc"
+	# Add the package's bin directory to the PATH.
+	echo "PATH=\"${EPREFIX}/opt/${PN}/bin\"" > "${T}/00fastqc" || die
+	doenvd "${T}/00fastqc"
+
 	dodoc README.txt RELEASE_NOTES.txt
 
 	# There is no fastqc.jar.  The output from the compilation is the set of
```

Comment:
Now I know that this is not your fault but could this message be cleared up? Do we still need it?
If it is actually important and needs action by the user, can you make this into a message that is given to the user in `pkg_postinst`?
Why do we not have a dependency on these packages `sci-libs/jhdf5` and `sci-biology/picard`? You seem to have dropped `sci-libs/jhdf5` in the update?
Give the fact that we don't have tests, we manually need to make sure we are installing the proper dependencies.
Seeing that you haven't made a version bump, possibly upstream has not made any new releases, so it is unlikely that we are now magically compatible with the newer version of the codebase. Is it possible (though unlikely) that upstream has changed to a new url?
I know these are a lot of questions but unfortunately with the lack of tests the only way I can know anything is by voicing it out :(

Reply:
* By "this message" you mean the long comment below the highlighted code? Yes, I think it can be cleared up. It was a note for a future maintainner (me) from the previous maintainer (mmokrejs).
* I have dropped the dependency on sci-libs/jhdf5 and sci-biology/picard because they are bundled with the package itself, so that fastqc is using its own copy of them.
* Yes, to "manually make sure we are installing the proper dependencies" I install the package into a Prefix environment and check that it runs for a sample input. For jhdf5 and picard, I also see that their .jar files are indeed included as promised, and get installed with the package, and that the main program adds them to the java's CLASSPATH.
* I believe upstream did make new releases, but I am trying to follow the motto of "make many small changes, rather than one huge one". Small changes are easier to review, and easier to not make mistakes in. I promise I will make a version bump separately.
* "Voicing it out" works for me! Really appreciate it.

Reply:
This is amazing though, they are using a perl script to invoke a java binary...
Absolutely amazing...

Reply:
@anton-molyboha Indeed, 0.11.9 is out. The HDF5 is necessary to work with Oxford nanopore FAST5 files, IMO.

Reply:
@mmokrejs Thanks. It's a moon-shot, but do you happen, by any chance, have an example of a FAST5 file I could use for testing?

Reply:
Or a .sam file...

Reply:
Try to fetch some from NCBI SRA. Maybe by default you get only FASTQ format but it should be possible to fetch some using sra-toolkit to obtain the original FAST5-formatted data.

https://www.ncbi.nlm.nih.gov//sra?term=(nanopore)%20NOT%20cluster_dbgap%5BPROP%5D

Alternatively, here you have plenty demo data:
https://github.com/nanopore-wgs-consortium/NA12878

Some more background info:
https://simpsonlab.github.io/2017/02/27/packing_fast5/

Reply:
Haha, to download from NCBI SRA I would have to fix the sra-toolkit package first, which is also broken! I will try the nanopore-wgs-consortium link you gave.